type/src/ai-card.ts

```diff
@@ -0,0 +1,197 @@
+
+  /**
+   * Holds all trust, compliance, and identity information.
+   */
+  trust: Trust;
+  
+  /**
+   * An array of all interaction protocols this agent supports.
+   * This is the core of the extensibility model.
+   */
+  services: BaseService[];
+
+  /**
+   * An ISO 8601 timestamp of when the agent was first published.
+   */
+  createdAt: string; 
+
+  /**
+   * An ISO 8601 timestamp of when this card was last updated.
+   */
+  updatedAt: string; 
+
+  /**
+   * An open "black box" for any other non-standard metadata.
+   */
+  metadata?: Record<string, any>;
+}
+// --8<-- [end:AICard]
+
+// --- CORE COMPONENTS ---
+
+// --8<-- [start:Publisher]
+/**
+ * Defines the entity (company, individual) that published the agent.
+ */
+export interface Publisher {
+  /**
+   * A verifiable ID for the publisher, e.g., a DID or organization ID.
+   */
+  id: string;
+
+  /**
+   * The human-readable name of the publisher.
+   */
+  name: string;
+
+  /**
+   * A URL to a verifiable credential (e.g., JWT) proving the publisher's identity.
+   */
+  attestation?: string;
+}
+// --8<-- [end:Publisher]
+
+// --8<-- [start:Identity]
+/**
+ * A verifiable identity for the agent or publisher.
+ */
+export interface Identity {
+  /**
+   * The type of identity (e.g., "did", "spiffe").
+   */
+  type: string;
+
+  /**
+   * The identity string itself.
+   */
+  id: string;
+}
+// --8<-- [end:Identity]
+
+// --8<-- [start:Trust]
+/**
+ * Defines the security, identity, and compliance posture of the agent.
+ */
+export interface Trust {
+  // NOTE: 'identity' and other fields can be added here later.
+  
+  /**
+   * A list of compliance or other attestations.
+   */
+  attestations?: Attestation[];
+}
+// --8<-- [end:Trust]
+
+// --8<-- [start:Attestation]
+/**
+ * A single compliance, security, or custom attestation.
+ */
+export interface Attestation {
+  /**
+   * The type of attestation (e.Go., "SOC2", "HIPAA", "CustomBadge").
+   */
+  type: string;
+
+  /**
+   * (Low-Trust) A URL to a simple JSON "badge" file.
+   */
+  badgeUrl?: string; 
+
+  /**
+   * (High-Trust) A URL to a verifiable credential (e.g., a JWT or PDF report).
+   * Use this when the credential is large or hosted remotely.
+   */
+  credentialUrl?: string; 
+
+  /**
+   * (High-Trust) The embedded, base64-encoded credential itself (e.g., a JWT).
+   * Use this for self-contained attestations (no extra network call).
+   */
+  credentialValue?: string;
+}
+// --8<-- [end:Attestation]
+
+// --8<-- [start:BaseService]
+/**
+ * A generic representation of an interaction service.
+ * All specific service types extend this.
+ */
+export interface BaseService {
+  /**
+   * A custom protocol identifier string (e.g., "a2a", "mcp", "foo").
+   */
+  type: string;
+  
+  /**
+   * A human-readable name for this specific service endpoint.
+   * For example, "Travel Agent A2A Endpoint".
+   */
+  name: string;
+
+  /**
+   * The full URL to the service endpoint.
+   */
+  endpoint: string;
```

Comment:
A2A has multiple URLs for a single "AI service".  I don't think people are going to want to duplicate the skills definitions for each protocol binding.